test-js/test-utils.js

```diff
@@ -0,0 +1,247 @@
+  getCharFromUnicode,
+  guessCharFromGlyphName,
+  fileNameExtension,
+  arrayExtend,
+} from "../src/fontra/client/core/utils.js";
+const expect = chai.expect;
+
+describe("objectsEquals", () => {
+  it("falsy values", () => {
+    expect(objectsEqual(null, null)).equals(true);
+    expect(objectsEqual(undefined, null)).equals(false);
+    expect(objectsEqual("", "")).equals(true);
+    expect(objectsEqual("", null)).equals(false);
+  });
+  it("with things inside", () => {
+    expect(objectsEqual({}, { a: 1 })).equals(false);
+    expect(objectsEqual({ a: 1 }, {})).equals(false);
+    expect(objectsEqual({ a: 1 }, { a: 2 })).equals(false);
+    expect(objectsEqual({ a: 2 }, { a: 2 })).equals(true);
+  });
+  it("thing in their prototype", () => {
+    // making sure only objects own properties checked
+    class Thing {
+      constructor(value) {
+        this.b = value;
+      }
+    }
+    Thing.prototype.a = 1;
+    expect(objectsEqual(new Thing(1), { b: 1 })).equals(true);
+    expect(objectsEqual(new Thing(1), { b: 1, a: 1 })).equals(false);
+  });
+});
+
+describe("capitalizeFirstLetter", () => {
+  it("basic functionality", () => {
+    expect(capitalizeFirstLetter("sam")).equals("Sam");
+    expect(capitalizeFirstLetter("Sam")).equals("Sam");
+  });
+  it("with spaces prefixed", () => {
+    expect(capitalizeFirstLetter(" sam")).equals(" sam");
+  });
+});
+
+describe("hyphenatedToCamelCase", () => {
+  it("should camelize", () => {
+    expect(hyphenatedToCamelCase("test-case")).equals("testCase");
+  });
+  it("should not delete the hypen when the second part is not a lowercase letter", () => {
+    expect(hyphenatedToCamelCase("test-1")).equals("test-1");
+  });
+});
+
+describe("modulo", () => {
+  it("should return the remaining when divide", () => {
+    expect(modulo(12, 5)).equals(2);
+  });
+  it("python behavior of modulus when mod a minus value", () => {
+    expect(modulo(-3, 5)).equals(2);
+  });
+});
+
+describe("boolInt", () => {
+  it("1 for truthy", () => {
+    expect(boolInt(true)).equals(1);
+    expect(boolInt([])).equals(1);
+    expect(boolInt(1)).equals(1);
+    expect(boolInt({})).equals(1);
+  });
+  it("0 for falsy", () => {
+    expect(boolInt(false)).equals(0);
+    expect(boolInt("")).equals(0);
+    expect(boolInt(0)).equals(0);
+    expect(boolInt(null)).equals(0);
+    expect(boolInt(undefined)).equals(0);
+  });
+});
+
+describe("reversed", () => {
+  it("reverse an iterator", () => {
+    const numbers = [1, 2, 3];
+    const numbersreversed = reversed(numbers);
+    expect(numbersreversed.next().value).equals(3);
+    expect(numbersreversed.next().value).equals(2);
+    expect(numbersreversed.next().value).equals(1);
+  });
+});
+
+describe("enumerate", () => {
+  it("enumerate an array, enumeration start with 0 by default", () => {
+    const numbers = [1, 2, 3];
+    const numbersenumerated = enumerate(numbers);
+    expect(numbersenumerated.next().value[0]).equals(0);
```

Comment:
This doesn't test whether enumerate returns the correct _value_, only the index. Unpacking to an array should help, as you can then compare against `[[0, 1], [1, 2], [2, 3]]` using `expect(...).to.deep.equal(...)`.